core/src/context.hpp

```diff
@@ -54,3 +54,196 @@ json & operator << (json &, km_core_context_item const &);
+ *                      May be null to request size calculation.
+ * @param buf_size      A pointer to the result variable: The size of the
+ *                      supplied buffer in codeunits if `buf` is given. On
+ *                      return will be the size required if `buf` is null.
+ */
+km_core_status
+context_items_to_utf8(km_core_context_item const *item,
+                              char *buf,
+                              size_t *buf_size);
+
+/**
+ * Convert a context item array into a UTF-32 encoded string placing it into
+ * the supplied buffer of specified size, and return the number of codepoints
+ * actually used in the conversion. If null is passed as the buffer the
+ * number of codepoints required is returned. Any markers in the context will
+ * not be included in the output buffer.
+ *
+ * @return km_core_status
+ *         * `KM_CORE_STATUS_OK`: On success.
+ *         * `KM_CORE_STATUS_INVALID_ARGUMENT`: If non-optional parameters are
+ *           null.
+ *         * `KM_CORE_STATUS_INSUFFICENT_BUFFER`: If the buffer is not large
+ *           enough. `buf_size` will contain the space required. The contents
+ *           of the buffer are undefined.
+ *
+ * @param context_items A pointer to the start of an array
+ *                      `km_core_context_item`. Must be terminated with a type
+ *                      of `KM_CORE_CT_END`.
+ * @param buf           A pointer to the buffer to place the UTF-32 string into.
+ *                      May be null to request size calculation.
+ * @param buf_size      A pointer to the result variable: The size of the
+ *                      supplied buffer in codepoints if `buf` is given. On
+ *                      return will be the size required if `buf` is null.
+ */
+km_core_status
+context_items_to_utf32(km_core_context_item const *item,
+                              km_core_usv *buf,
+                              size_t *buf_size);
+/**
+ *
+ * Copies all items in the context into a new array and returns the new array.
+ * This must be disposed of by caller using `km_core_context_items_dispose`.
+ *
+ * @return km_core_status
+ *         * `KM_CORE_STATUS_OK`: On success.
+ *         * `KM_CORE_STATUS_INVALID_ARGUMENT`: If non-optional parameters are
+ *           null.
+ *         * `KM_CORE_STATUS_NO_MEM`: In the event not enough memory can be
+ *           allocated for the output buffer.
+ *
+ * @param context_items A pointer to the start of an array
+ *                      `km_core_context_item`.
+ *
+ * @param out           A pointer to the result variable: A pointer to the start
+ *                      of the `km_core_context_item` array containing a copy of
+ *                      the context. Terminated with a type of `KM_CORE_CT_END`.
+ *                      Must be disposed of with
+ *                      `km_core_context_items_dispose`.
+ */
+km_core_status
+context_get(km_core_context const *context_items,
```

Comment:
Shouldn't the parameter be named `context`?

```suggestion
context_get(km_core_context const *context,
```